app/controllers/saved_pages_controller.rb

```diff
@@ -0,0 +1,72 @@
+class SavedPagesController < ApplicationController
+  include AuthenticatedApiConcern
+
+  before_action :check_saved_page_exists, only: %i[destroy show]
+
+  # GET /api/saved_pages
+  def index
+    @saved_pages = SavedPage.where(oidc_user_id: user.id)
+    render_api_response(saved_pages: present_pages(@saved_pages))
+  end
+
+  # GET /api/saved_pages/:page_path
+  def show
+    render_api_response(saved_page: present_page(saved_page))
+  end
+
+  # POST /api/saved_pages
+  def create
+    saved_page = SavedPage.new(oidc_user_id: user.id, page_path: params[:page_path])
+
+    if saved_page.save
+      render_api_response(saved_page: present_page(saved_page))
+    else
+      handle_create_saved_page_errors(saved_page)
+    end
+  end
+
+  # DELETE /api/saved_pages
+  def destroy
+    return head :no_content if saved_page.destroy!
+
+    raise ApiError::SavedPageNotDestroyed, { page_path: params[:page_path] }
+  end
+
+private
+
+  def handle_create_saved_page_errors(saved_page)
+    error_types = saved_page.errors.map(&:type)
+
+    if error_types.include?(:taken)
+      raise ApiError::SavedPageDuplicate, { page_path: params[:page_path] }
+    end
+
+    if error_types.include?(:blank)
+      raise ApiError::SavedPagePathRequired, { page_path: params[:page_path] }
+    end
+
+    if error_types.any?
+      raise ApiError::SavedPageNotPersisted, { page_path: params[:page_path] }
+    end
+  end
+
+  def present_pages(saved_pages)
+    saved_pages.map { |saved_page| present_page(saved_page) }
+  end
+
+  def present_page(saved_page)
+    { "page_path" => saved_page.page_path }
+  end
```

Comment:
Should this instead be a `to_hash` on the `SavedPage` model?

Reply:
I like the `SavedPage#to_hash` suggestion. With that implemented, we can go for the nice
```ruby
def index
  render_api_response(saved_pages: user.saved_pages.map(&:to_hash))
end
```
and we don't need the two private methods.

Reply:
If we did though wouldn't it include other attribtues from a page?
Currently the API docs have been fairly specific about whats returned (which might be OTT?)
But so I'd wanted to constrain this and make it deliberate to add new fields so we remember to updat the API docs in the same commit?

Reply:
> If we did though wouldn't it include other attribtues from a page?

Depends on what we make `to_hash` return

Reply:
Based on the `present_page(saved_page)` method, `SavedPage#to_hash` can be

```ruby
def to_hash
  { "page_path" => page_path }
end
```

Reply:
humm, `to_hash` might confuse with the `to_h` method that would do something similar but more potentially?
I went with `present` as `presenters` are an established pattern for formatting and filtering return values I'd thought?

Reply:
We already use `to_hash` in a few places ([here in account-api](https://github.com/alphagov/account-api/blob/056066400e0edd8be98fbf65cc76159027d242d5/app/lib/account_session.rb#L68-L75) and [here in govuk-account-manager-prototype](https://github.com/alphagov/govuk-account-manager-prototype/blob/7858b55afcddc2f855d909abd051838d95f21f13/app/models/ephemeral_state.rb#L7-L13)) so it would be consistent with what we do already. But I don't see anything wrong with `present` either.

Reply:
I think I'd prefer something based on "presenter" than just "present", as there's already a `present?` method